### PR TITLE
Homogenize levels/eigenfunctions terminology

### PR DESCRIPTION
**Pull request by stoprightthere**

### CODE REVIEW

File: geometric_kernels/kernels/geometric_kernels.py [hunk 1]
```diff
@@ -15,16 +15,21 @@
...
     eigenfunctions of the Laplace-Beltrami operator and :math:`S(\cdot)` the
     spectrum of the stationary kernel. The eigenvalues and eigenfunctions belong
-    to the `SpaceWithEigenDecomposition` instance.
+    to the `DiscreteSpectrumSpace` instance.
+
+    We refer to sets of :math:`\{\phi_{ij}\}_{j=0}^{K_i}` as
+    "levels". In many practical spaces, like the sphere, we can employ
+    _addition theorems_ to efficiently compute the sums over levels
+    :math:`\sum_{j=0}^{K_i} \phi_{ij}(x) \phi_{ij}(x')`. A level may
+    consist of one eigenfunction.
```
Comment (vabor112) [inline geometric_kernels/kernels/geometric_kernels.py:32]: Maybe
-  "and eigenfunctions of the Laplace-Beltrami operator" -> " and eigenfunctions of the Laplace-Beltrami operator :math:`\Delta` such that :math:`\Delta \phi_{ij} = \lambda_i \phi_{ij}`".
- " We refer to sets of :math:`\{\phi_{ij}\}_{j=0}^{K_i}` as "levels". In many practical spaces, like the sphere, we can employ _addition theorems_ to efficiently compute the sums over levels :math:`\sum_{j=0}^{K_i} \phi_{ij}(x) \phi_{ij}(x')`. A level may consist of one eigenfunction."
  ->
   "We refer to the pairs :math:`(\lambda_i, G_i(\cdot, \cdot')` where :math:`G_i(\cdot, \cdot') = \sum_{j=0}^{K_i} \phi_{ij}(\cdot) \phi_{ij}(\cdot')` as "levels". For many spaces, like the sphere, we can employ _addition theorems_ to efficiently compute :math:`G_i(\cdot, \cdot')` without calculating the individual :math:`\phi_{ij}`. Note that although  :math:`\phi_{ij}` are guaranteed to correspond to the same eigenvalue :math:`\lambda_i`, the meaning of :math:`K_i` can vary, it does not have to be equal to the dimension of the whole eigenspace corresponding to :math:`\lambda_i`. For example, for `Graph` and `Mesh` it is always equal to 1 with eigenvalues having multiplicity greater than one giving rise to multiple levels with equal :math:`\lambda_i` but different :math:`G_i(\cdot, \cdot')`.".

File: geometric_kernels/kernels/geometric_kernels.py [hunk 2]
```diff
@@ -33,24 +38,25 @@ class MaternKarhunenLoeveKernel(BaseGeometricKernel):
...
+        num_levels: int,
     ):
         r"""
         :param space: Space providing the eigenvalues and eigenfunctions of
             the Laplace-Beltrami operator.
         :param nu: Determines continuity of the Mat\'ern kernel. Typical values
             include 1/2 (i.e., the Exponential kernel), 3/2, 5/2 and +\infty
             `np.inf` which corresponds to the Squared Exponential kernel.
-        :param num_eigenfunctions: number of eigenvalues and functions to include
+        :param num_levels: number of eigenvalues and levels to include
```
Comment (vabor112) [inline geometric_kernels/kernels/geometric_kernels.py:55]: Maybe "eigenvalues and levels" -> "levels"?

File: geometric_kernels/spaces/base.py [hunk 3]
```diff
@@ -39,14 +39,14 @@ class DiscreteSpectrumSpace(Space):
     @abc.abstractmethod
     def get_eigenfunctions(self, num: int) -> Eigenfunctions:
         """
-        First `num` eigenfunctions of the Laplace-Beltrami operator
+        Eigenfunctions of the first `num` levels of the Laplace-Beltrami operator.
```
Comment (vabor112) [inline geometric_kernels/spaces/base.py:42]: "Eigenfunctions of the first `num` levels of the Laplace-Beltrami operator." -> "Eigenfunctions of the Laplace-Beltrami operator corresponding to the first `num` levels."

File: geometric_kernels/spaces/base.py [hunk 4]
```diff
@@ -39,14 +39,14 @@ class DiscreteSpectrumSpace(Space):
...
         """
         raise NotImplementedError
 
     @abc.abstractmethod
     def get_eigenvalues(self, num: int) -> B.Numeric:
         """
-        First `num` eigenvalues of the Laplace-Beltrami operator
+        First `num` eigenvalues of the Laplace-Beltrami operator.
```
Comment (vabor112) [inline geometric_kernels/spaces/base.py:49]: With or without repetitions? Or corresponding to the first `num` levels?

File: geometric_kernels/spaces/product.py [hunk 5]
```diff
@@ -260,7 +260,7 @@ def dim_of_eigenspaces(self):
 
 
 class ProductDiscreteSpectrumSpace(DiscreteSpectrumSpace):
-    def __init__(self, *spaces: DiscreteSpectrumSpace, num_eigen: int = 100):
+    def __init__(self, *spaces: DiscreteSpectrumSpace, num_eigenvalues: int = 32):
```
Comment (vabor112) [inline geometric_kernels/spaces/product.py:263]: which ones, unique, with repetitions, or is this actually the number of levels (which is, in general, neither of these)?

**Review by vabor112 — CHANGES_REQUESTED** [review]

File: geometric_kernels/kernels/geometric_kernels.py [hunk 6]
```diff
@@ -15,16 +15,21 @@
...
-    This class approximates a kernel by the finite feature decomposition using
+    r"""This class approximates a kernel by the finite feature decomposition using
     its Laplace-Beltrami eigenfunctions and eigenvalues [1, 2].
 
-    .. math:: k(x, x') = \sum_{i=0}^{M-1} S(\sqrt\lambda_i) \phi_i(x) \phi_i(x'),
+    .. math:: k(x, x') = \sum_{i=0}^{M-1} S(\sqrt\lambda_i) \sum_{j=0}^{K_i} \phi_{ij}(x) \phi_{ij}(x'),
 
-    where :math:`\lambda_i` and :math:`\phi_i(\cdot)` are the eigenvalues and
+    where :math:`\lambda_i` and :math:`\phi_{ij}(\cdot)` are the eigenvalues and
     eigenfunctions of the Laplace-Beltrami operator and :math:`S(\cdot)` the
```
Comment (stoprightthere) [inline geometric_kernels/kernels/geometric_kernels.py:24]:
```suggestion
    eigenfunctions of the Laplace-Beltrami operator such that :math:`\Delta \phi_{ij} = \lambda_i \phi_{ij}` and :math:`S(\cdot)` the
```

**Pull request merged**